contracts/rbn-staking/IncentivizedVotingLockup.sol

```diff
@@ -676,12 +1006,166 @@ contract IncentivisedVotingLockup is
+    address _addr,
+    uint256 _block,
+    uint128 _nextExpiry
+  )
+    internal
+    view
+    returns (
+      uint256,
+      int128,
+      uint256,
+      int128
+    )
+  {
+    require(_block < block.number, "sRBN::getPriorVotes: not yet determined");
+
+    uint32 nCheckpoints = numCheckpoints[_addr];
+    if (nCheckpoints == 0 || _nextExpiry == 0) {
+      return (0, 0, 0, 0);
+    }
+
+    Boost memory cp;
+
+    // First check most recent balance
+    if (_boost[_addr][nCheckpoints - 1].fromBlock <= _block) {
+      cp = _boost[_addr][nCheckpoints - 1];
+      return (
+        cp.delegatedBias,
+        cp.delegatedSlope,
+        cp.receivedBias,
+        cp.receivedSlope
+      );
+    }
+
+    // Next check implicit zero balance
+    if (_boost[_addr][0].fromBlock > _block) {
+      return (0, 0, 0, 0);
+    }
+
+    uint32 lower = 0;
+    uint32 upper = nCheckpoints - 1;
+    while (upper > lower) {
+      uint32 center = upper - (upper - lower) / 2; // ceil, avoiding overflow
+      cp = _boost[_addr][center];
+      if (cp.fromBlock == _block) {
+        return (
+          cp.delegatedBias,
+          cp.delegatedSlope,
+          cp.receivedBias,
+          cp.receivedSlope
+        );
+      } else if (cp.fromBlock < _block) {
+        lower = center;
+      } else {
+        upper = center - 1;
+      }
+    }
+    cp = _boost[_addr][lower];
+    return (
+      cp.delegatedBias,
+      cp.delegatedSlope,
+      cp.receivedBias,
+      cp.receivedSlope
+    );
+  }
+
+  function _calcBiasSlope(
+    int256 _x,
+    int256 _y,
+    int256 _expireTime
+  ) internal pure returns (int128 slope, uint256 bias) {
+    // SLOPE: (y2 - y1) / (x2 - x1)
+    // BIAS: y = mx + b -> y - mx = b
+    slope = SafeCast.toInt128(-_y / (_expireTime - _x));
+    bias = uint256(_y - slope * _x);
+  }
+
+  function calcBoostBiasSlope(address _delegator)
+    external
+    view
+    returns (int128, uint256)
+  {
+    if (delegates[_delegator] == address(0)) {
+      return (0, 0);
+    }
+
+    uint32 nCheckpointsDelegator = numCheckpoints[_delegator];
+    uint128 expireTime =
+      _boost[msg.sender][nCheckpointsDelegator - 1].nextExpiry;
+
+    if (expireTime == 0) {
+      return (0, 0);
+    }
+
+    uint256 delegatedBias =
+      _boost[_delegator][nCheckpointsDelegator - 1].delegatedBias;
+    int128 delegatedSlope =
+      _boost[_delegator][nCheckpointsDelegator - 1].delegatedSlope;
+
+    // delegated boost will be positive, if any of circulating boosts are negative
+    // we have already reverted
+    int256 delegatedBoost =
+      delegatedSlope * int256(block.timestamp) + int256(delegatedBias);
+    int256 y = int256(uint256(getCurrentVotes(_delegator))) - delegatedBoost;
+    require(y > 0, "No boost");
+
+    (int128 slope, uint256 bias) =
+      _calcBiasSlope(int256(block.timestamp), y, int256(block.timestamp));
+    require(slope < 0, "invalid slope");
+
+    return (slope, bias);
+  }
+
```

Comment:
compressed [delegated_boost()](https://github.com/curvefi/curve-veBoost/blob/0e51be10638df2479d9e341c07fafa940ef58596/contracts/VotingEscrowDelegation.vy#L857) and [received_boost()](https://github.com/curvefi/curve-veBoost/blob/0e51be10638df2479d9e341c07fafa940ef58596/contracts/VotingEscrowDelegation.vy#L872)